R/FilterSpectroOut.R

```diff
@@ -0,0 +1,659 @@
+#' @param annotS A data.frame object containing sample annotation. Rows are
+#' samples and must match to columns of \code{quantityList}.
+#' If \code{filterNA='byCondition'} it needs to contain column providing the
+#' conditions/groups.
+#' @param filterNA A character variable defining type of NA filtering applied.
+#' Can be set to c('all', 'byCondition' or 'none').
+#' \itemize{
+#'   \item 'all': Filtering of NAs will be applied on all samples, independent
+#'   of their condition. \code{maxNA} defines the number of NAs allowed per
+#'   quantity over all samples.
+#'   \item 'byCondition': Filtering of NAs will be applied on samples based on
+#'   their condition.  \code{maxNA} defines the number of NAs allowed per
+#'   quantity over all samples of each individual condition. It is advised to
+#'   only use 'byCondition', if the condition you are providing is a
+#'   categorical variable and not continuous.
+#'   \item 'none': No NA filtering will be applied. This my cause issues if
+#'   running RUV and/or contrast models subsequently.
+#'   }
+#' @param maxNAs A numeric value, defining maximal number of NAs allowed.
+#' Default is '0'.
+#' @param infoCondition A character string or numeric giving the column name
+#' or column number of \code{annotS} in which condition is provided. Required,
+#' if \code{filterNA = 'byCondition'}.
+#' Default is 'Condition'.
+#' @param nameIDQuant A character string giving column of \code{annotPP} which
+#' matches to the row.names of \code{quantityList} or \code{quantityMatrix}.
+#' Default is 'quantID'.
+#' @param nameProtQuant A character string giving column of \code{annotPP} which
+#' protein names are provided. If a peptides matches to the same protein several
+#' times, the protein name should be provided each time, separated by ','. If a
+#' peptide maps to the multiple proteins, these different proteins can be
+#' provided by separating them with ';'.
+#' Default is 'Protein'.
+#' @param infoTryptic A character string or numeric giving the column name or
+#' column number of \code{annotPP} in which digest type is provided.
+#' Default is 'isTryptic'.
+#' @param nameFT A character string defining how full-tryptic peptides are
+#' annotated in the \code{infoTryptic} colunm in \code{annotPP}. Only peptides
+#' annotated accordingly will remain in the data if \code{filterTryptic} is set
+#' to 'TRUE'.
+#' Default is 'Specific'.
+#'
+#' @return Returns a list of matrices containing half-tryptic LiP quantities of
+#' interest (e.g. peptide, modified peptide, precursor), the best matching
+#' full-tryptic TrP quantities of interest (e.g. peptide, modified peptide,
+#' precursor) and corresponding TrP protein quantities. The row names in all
+#' matrices correspond to the half-tryptic LiP quantity ID.
+filterForHTonly <- function(quantityList, annotPP, annotS, filterNA, maxNAs,
+                            infoCondition, nameIDQuant, nameProtQuant,
+                            infoTryptic, nameFT){
+
```

Comment:
same as with the function above especially as you provide defaults in the comments